### ISSUE TITLE
Performance Analytics Dashboard & Continuous Improvement

### ISSUE THREAD

**Issue by BdM-15** [BdM-15/govcon-capture-vibe#22]
## Strategic Benefits: Performance Analytics

Comprehensive analytics for system performance, user behavior, and continuous improvement through data-driven optimization.

#### 1. A/B Testing for Prompts
- **Prompt Variation Testing**: Compare extraction accuracy across prompt versions
- **Entity-Specific Optimization**: Test prompts per entity type (REQUIREMENT, DELIVERABLE, etc.)
- **Statistical Significance**: Automated analysis of prompt performance differences
- **Related Issues**: 
  - Issue #9: Entity Extraction Prompt Refinement
  - Issue #14: Prompt Compression Strategy

#### 2. Processing Optimization Metrics
- **Per-RFP Cost Tracking**: Token usage, API costs, processing time by document
- **Chunk Efficiency Analysis**: Optimal chunk sizes for different document types
- **LLM Call Reduction**: Track improvements from optimization issues
- **Related Issues**:
  - Issue #14: Prompt Compression (~40% token reduction target)
  - Issue #15: Remove Redundant Algorithms
  - Issue #18: Optimal Chunk Size Analysis

**Example Dashboard Metrics**:
```
┌─────────────────────────────────────────────────────────┐
│ Processing Cost Analytics (Last 30 Days)                │
├─────────────────────────────────────────────────────────┤
│ Total RFPs Processed: 12                                │
│ Average Cost/RFP: $3.42 (target: $2.00)                │
│ Average Processing Time: 45 min (target: 30 min)        │
│ Total API Tokens: 24.5M                                 │
│                                                         │
│ Cost Breakdown:                                         │
│   Extraction Phase: 68% ($2.33/RFP)                    │
│   Post-Processing: 22% ($0.75/RFP)                     │
│   Query Serving: 10% ($0.34/RFP)                       │
└─────────────────────────────────────────────────────────┘
```

#### 3. Extraction Accuracy Metrics
- **Entity Extraction Precision/Recall**: Validated against human review
- **Relationship Accuracy**: Section L↔M mapping correctness
- **Error Pattern Analysis**: Common extraction failures by entity type
- **Related Issues**:
  - Issue #6: Pydantic Schema Extraction
  - Issue #7: Verbatim Extraction
  - Issue #11: Pydantic Enforcement for Post-Processing

**Accuracy Dashboard**:
```
┌─────────────────────────────────────────────────────────┐
│ Extraction Quality Metrics                              │
├─────────────────────────────────────────────────────────┤
│ Entity Type       │ Precision │ Recall │ F1 Score      │
│───────────────────┼───────────┼────────┼───────────────│
│ REQUIREMENT       │ 94.2%     │ 89.1%  │ 91.6%         │
│ EVALUATION_FACTOR │ 97.8%     │ 95.3%  │ 96.5%         │
│ DELIVERABLE       │ 88.4%     │ 82.7%  │ 85.4%         │
│ CLAUSE            │ 99.1%     │ 98.7%  │ 98.9%         │
│ SUBMISSION_INST   │ 93.5%     │ 91.2%  │ 92.3%         │
└─────────────────────────────────────────────────────────┘
```

#### 4. User Satisfaction Tracking
- **Query Response Quality**: User feedback on answer relevance
- **Workflow Efficiency**: Time from RFP upload to actionable insights
- **Feature Usage Analytics**: Which capabilities most valuable
- **Error/Confusion Tracking**: Common user pain points

**User Feedback Integration**:
```python
# Feedback loop for training data collection
# Reference: docs/future_features/performance/FINE_TUNING_ROADMAP.md

@app.post("/feedback")
async def submit_feedback(
    query_id: str,
    rating: Literal["good", "bad", "partial"],
    correction: Optional[str] = None
):
    """Collect user feedback for training data pipeline."""
    # Store for fine-tuning dataset (Issue #19)
    await training_data_collector.log_feedback(
        query_id=query_id,
        rating=rating,
        correction=correction
    )
```

### Training Data Pipeline Integration

This issue connects directly to Issue #19 (Fine-Tuned SLM Strategy):

**Data Collection Points**:
1. **Extraction Corrections**: User fixes to entity extraction → training data
2. **Query Improvements**: Better answers from user refinement → preference data
3. **Relationship Validation**: Human-verified L↔M mappings → ground truth
4. **Quality Scores**: Per-extraction confidence ratings

**Reference**: `docs/future_features/performance/FINE_TUNING_ROADMAP.md`
- Phase 1: Training Data Collection (500-1000 examples)
- Error Pattern Analysis for prompt improvement

### Technical Implementation

**Logging Infrastructure**:
```python
# Structured logging for analytics
import structlog

logger = structlog.get_logger()

async def log_extraction_metrics(
    rfp_id: str,
    entity_type: str,
    tokens_used: int,
    extraction_time: float,
    entities_found: int
):
    logger.info(
        "extraction_complete",
        rfp_id=rfp_id,
        entity_type=entity_type,
        tokens=tokens_used,
        time_seconds=extraction_time,
        entity_count=entities_found
    )
```

**Dashboard Technology Options**:
- Streamlit (quick prototyping)
- Grafana + Prometheus (production monitoring)
- Custom React dashboard (full control)

### Dependencies

| Dependency | Status | Related Issue |
|------------|--------|---------------|
| Structured Logging | Needed | New |
| Metrics Database | Needed | PostgreSQL (#20) |
| Training Data Pipeline | Planned | #19 |
| Prompt Testing Framework | Needed | #9, #14 |

### Estimated Effort
- Logging Infrastructure: 15-20 hours
- Metrics Collection: 20-30 hours
- Dashboard Development: 30-40 hours
- A/B Testing Framework: 25-35 hours
- **Total**: 90-125 hours

### Success Metrics
- Real-time visibility into processing costs
- Extraction accuracy tracked per entity type
- Prompt improvements measurable via A/B testing
- User satisfaction feedback integrated into training pipeline

### Related Issues
- Issue #9: Entity Extraction Prompt Refinement
- Issue #14: Prompt Compression Strategy
- Issue #15: Remove Redundant Algorithms
- Issue #18: Optimal Chunk Size Analysis
- Issue #19: Fine-Tuned SLM Strategy (training data pipeline)

---

**Milestone**: Phase 8+ (can start before PostgreSQL)
**Priority**: Medium-High (enables data-driven optimization)
